build.gradle

```diff
@@ -113,6 +124,22 @@ jar {
     from sourceSets.generated.output
 }
 
+def getGitHash = { ->
+    def stdout = new ByteArrayOutputStream()
+    exec {
+        commandLine 'git', 'rev-parse', '--short', 'HEAD'
```

Comment:
Re: these two assumptions:

- I think it is fine. Locally, running this command from anywhere in the project or its subdirectories will lead to the correct repository being referenced. In Travis, the working directory seems to be the top level directory of the project so we have a guarantee that this will be referencing the correct repository there.
- Since this is not communicating with a remote server, just checking on the current state of the local git repo, there will be no authentication info required.